webview/src/setup/components/ProjectUninitialized.tsx

```diff
@@ -1,28 +1,83 @@
-import { VSCodeButton } from '@vscode/webview-ui-toolkit/react'
 import React from 'react'
 import { EmptyState } from '../../shared/components/emptyState/EmptyState'
+import { Button } from '../../shared/components/button/Button'
 
-interface ProjectUninitializedProps {
-  initializeProject: () => void
+const Header: React.FC = () => <h1>DVC is not initialized</h1>
+
+interface GitUninitializedProps {
+  canGitInitialize: boolean | undefined
+  initializeGit: () => void
 }
 
-export const ProjectUninitialized: React.FC<ProjectUninitializedProps> = ({
-  initializeProject
+const GitIsPrerequisite: React.FC = () => (
+  <p>A Git repository is a prerequisite of project initialization</p>
+)
+
+const GitUninitialized: React.FC<GitUninitializedProps> = ({
+  canGitInitialize,
+  initializeGit
 }) => {
-  return (
-    <EmptyState>
-      <div>
-        <h1>DVC is not initialized</h1>
+  if (!canGitInitialize) {
+    return (
+      <EmptyState>
+        <Header />
+        <GitIsPrerequisite />
         <p>
-          The current workspace does not contain a DVC project. You can
-          initialize a project which will enable features powered by DVC. To
-          learn more about how to use DVC please read{' '}
-          <a href="https://dvc.org/doc">our docs</a>
+          The extension is unable to initialize a Git repository in this
+          workspace
         </p>
-        <VSCodeButton onClick={initializeProject}>
-          Initialize Project
-        </VSCodeButton>
-      </div>
+        <p>
+          Please open a different folder which contains no Git repositories or a
+          single existing Git repository at the root
+        </p>
+      </EmptyState>
+    )
+  }
+
+  return (
+    <EmptyState>
+      <Header />
+      <GitIsPrerequisite />
+      <Button onClick={initializeGit} text="Initialize Git" />
     </EmptyState>
   )
 }
+
+const DvcUninitialized: React.FC<{ initializeDvc: () => void }> = ({
+  initializeDvc
+}) => (
+  <EmptyState>
+    <Header />
+    <p>
+      The current workspace does not contain a DVC project. You can initialize a
+      project which will enable features powered by DVC. To learn more about how
+      to use DVC please read <a href="https://dvc.org/doc">our docs</a>
```

Comment:
I'm no grammar expert, but should this be: 

```suggestion
      to use DVC, please read <a href="https://dvc.org/doc">our docs</a>.
```

Reply:
Now that I look at it, a lot of the setup statements don't end with a period. Out of curiosity, is there a particular reason why?

Reply:
I don't have a good answer. Guess I'm going to spend some time today adding full stops to text.

Reply:
I think it is a spill-over from the titles